### ISSUE TITLE
Curious sortkey override behavior

### ISSUE THREAD

**Issue by ryanaustincarlson** [fishtown-analytics/dbt#2049]
### Describe the bug
We’ve run into an issue with defining sort config parameters in multiple places if both values are lists.

If…
1. a default sort config is defined in the dbt_project;
2. a model-specific sort parameter is defined; and
3. both sort configurations are lists

then the resulting sort key is a concatenation of all of the keys, at least on Redshift (not sure how platform-dependent this might be).

### Steps To Reproduce
Define a list of sort keys on a particular path in dbt_project
```yaml
models:
  project_name:
    events:
      sort: ['occurred_at', 'client']
```

Then, create a new model in that path (e.g., `project_name/events/cool_events_facts.sql`) and redefine sort-keys
```
{{
    config(
        sort=['day_of_week','version']
    )
}}
```

When the model is built, the resulting SQL concatenates all four values, something like this (on Redshift):
```SQL
create table
    events.cool_events_facts
    compound sortkey(day_of_week,version,occurred_at,client)
as (
    select *
    from whatever
);
```

### Expected behavior
My expectation is that the model-specific sort param overrides those in dbt_project, which is what happens when a string literal is used instead of a list. 

### Screenshots and log output
LMK if you need any more info!

### System information
**Which database are you using dbt with?**
- [ ] postgres
- [x] redshift
- [ ] bigquery
- [ ] snowflake
- [ ] other (specify: ____________)

**The output of `dbt --version`:**
```
installed version: 0.15.0
   latest version: 0.15.0

Up to date!
```

**The operating system you're using:**

macOS v10.14.6

**The output of `python --version`:**

Python 3.7.6

### Additional context
Our current workaround is to define the model-specific sort params in dbt_project, which seems to override as expected.

**Comment by drewbanin** [comment]
Thanks for the report @ryanaustincarlson! I was able to reproduce this pretty easily, but it was tough getting to the bottom of the actual issue!

I started working on a fix for this over here: https://github.com/fishtown-analytics/dbt/pull/2062. This PR should be shipped for our next minor release, v0.16.0.

**Comment by drewbanin** [comment]
Fixed by #2062
